### PR TITLE
Implement unit tests and consistent codestyle, fix syntax issues

### PR DESCRIPTION
**Pull request by EricHanLiu**
Unit tests using `Jest` and `Selenium`, small syntax errors fixed to remove Lint `JSHint` warnings, codestyle consistency enforced

Ensured compatibility with Travis and fixed build issues

Documented everything in README

### CODE REVIEW

File: source/js/plugins/pixel-wrapper.test.js [hunk 1]
```diff
@@ -3,7 +3,7 @@
 const {Builder, By, Key, until} = require('selenium-webdriver');
 const firefox = require('selenium-webdriver/firefox');
 // this is a Pixel.js job in Rodan
-const url = 'http://132.206.14.127:8000/interactive/bb128fdf-b30a-4322-ac87-a475ec5347ff/4245e880-5b05-4d11-89d3-f4128103c997/';
+const url = process.env.PIXEL_URL;
```
Comment (deepio) [inline source/js/plugins/pixel-wrapper.test.js]: This is better.

File: source/js/plugins/pixel-wrapper.test.js [hunk 2]
```diff
@@ -68,20 +68,60 @@ describe('Check Functionality', () => {
...
+    });
+    test('cancel progress bar works', async() => {
+        let cancelButton = await browser.findElement(By.id('cancel-export-div'));
+        const actions = browser.actions();
+        await actions.click(cancelButton).perform();
+
+        expect(await browser.findElements(By.id('pixel-export-div'))).toEqual([]); // empty list since not found
+    });
+});
+
```
Comment (deepio) [inline source/js/plugins/pixel-wrapper.test.js]: Try to avoid extra lines like this in your commits.

**Comment by deepio** [comment]
@EricHanLiu In the beginning, there are a few large commits where you try to fix multiple unrelated things, and then you change a small section later (fixing a fix, of a specific section of another fix). Try to have a cleaner git history because it makes it harder to track what you did during review. Keep your commits small, think about it as if you're limiting a commit to the length of something specific that you may need to 'revert' in the future. And you can also delete a pushed commit, with some restrictions obviously. Obviously i'm not asking you to revert all the changes and reorganize them now, that would be a huge waste of time.

- Normally i'd advise against making codestyle fixes, however, **in this project**, I'd accept it because it is following the linter that was setup by the original author. 

- What is the problem according to jshint for these?
https://github.com/DDMAL/pixel_wrapper/blob/008f2ea146bd948ed1ecda7ebfaebe21ee4d8aea/source/js/interpolate-animation.js#L34

`now` is undefined because its commented out here: 
https://github.com/DDMAL/pixel_wrapper/blob/e0750621228faab217c838114ba5c73f611dec06/source/js/interpolate-animation.js#L11-L22

Maybe they thought `now = () => { return performance.now(); };` looked ugly with two semicolons, but if that's the case I would have picked a different code style or (if eslint+jshint supports it) made some minor adjustments to the rules. 

https://github.com/DDMAL/pixel_wrapper/blob/008f2ea146bd948ed1ecda7ebfaebe21ee4d8aea/source/js/plugins/pixel-wrapper.js#L90
Is probably something similar.

File: source/js/plugins/pixel-wrapper.test.js [hunk 3]
```diff
@@ -97,7 +95,7 @@ describe('Check Functionality', () => {
         expect(await alert.getText()).toBe("You haven't created any select regions!");
         await alert.accept();
     });
-    test('drawing select region then submit to rodan creates progress bar', async() => {
+    test('drawing select region then submit to rodan creates progress bar', async () => {
```
Comment (deepio) [inline source/js/plugins/pixel-wrapper.test.js]: Try to catch these before committing them too.

File: source/js/plugins/pixel-wrapper.test.js [hunk 4]
```diff
@@ -80,13 +78,13 @@ describe('Check Functionality', () => {
         await actions.click(exportPNGButton).perform();
 
         // links should now be visible
-        let links = await browser.findElement(By.id('png-links-div'));
-        expect(links).toBeDefined();
+        let links = await browser.wait(until.elementLocated(By.id('png-links-div')), 1000);
```
Comment (deepio) [inline source/js/plugins/pixel-wrapper.test.js]: This is good.

**Comment by EricHanLiu** [comment]
@deepio Yeah the JSHint warning was because there weren't semicolons, so they commented that section out. In general I didn't change the codestyle of what was already written, just tried to fix the JSHint warnings wherever I could (missing semicolons, missing `break` statement).

Should I add the semicolons there and remove the `// jshint ignore` on the `now()`statements

File: .travis.yml [hunk 5]
```diff
@@ -1,22 +1,24 @@
 language: node_js
-node_js: 5  # Can't update to 6 until gulp does not depend on vinyl-fs 0.3.0
+node_js: 10  
+os: osx
+env:
+  - HOMEBREW_NO_AUTO_UPDATE="1"
 if: (branch = master AND type = (pull_request OR cron)) OR (branch = dev)
```
Comment (deepio) [inline .travis.yml]: This is never used.

File: README.md [hunk 6]
```diff
@@ -27,7 +27,7 @@ A wrapper to run [```Pixel.js```](https://github.com/DDMAL/Pixel.js) on top of [
       # Paths to other jobs
     )
     ```
-  - Increase `RODAN_RUNJOB_WORKING_USER_EXPIRY_SECONDS` (line 87) from `15` to `150000`.
+  - Increase `RODAN_RUNJOB_WORKING_USER_EXPIRY_SECONDS` (line 87) from `15` to `15000000`.
```
Comment (deepio) [inline README.md]: Is this really necessary?

**Comment by EricHanLiu** [comment]
Will try to move the tests to target a new server, only hosting once the tests are run (remove dependence on Rodan running)

File: gulpfile.js [hunk 7]
```diff
@@ -47,19 +47,19 @@ function lintGulpfile ()
...
 {
-	let divaConfig = Object.create(webpackConf)[0];
+	var divaConfig = Object.create(webpackConf)[0];
     webpack(divaConfig).run(done);
 }
 
 function server ()
 {
-    let devConfig = Object.create(webpackConf)[0];
+    var devConfig = Object.create(webpackConf)[0];
```
Comment (deepio) [inline gulpfile.js]: What was the error in jslint for these?

File: source/js/plugins/_filters.js [hunk 8]
```diff
@@ -15,10 +15,8 @@ export function addFilterToQueue (filter)
     }
 
     let fkeys = Object.keys(_filterQueue);
-    let flen = fkeys.length;
-    let flenScale = 1 / flen;
 
-    fkeys.map( (filt) => {
```
Comment (deepio) [inline source/js/plugins/_filters.js]: Is this really being used at all?

File: source/js/plugins/pixel-wrapper.js [hunk 9]
```diff
@@ -76,25 +76,26 @@ export class PixelWrapper
...
         this.pixelInstance.tools.currentTool = "rectangle";
 
         // Only create default layers once
-        if (this.layers.length !== 1) {
+        if (this.layers.length !== 1) 
             return;
-        }
 
-        let numLayers = numberInputLayers;
+        let numLayers = numberInputLayers; // jshint ignore:line
```
Comment (deepio) [inline source/js/plugins/pixel-wrapper.js]: What is the issue here?

File: source/js/plugins/pixel-wrapper.test.js [hunk 10]
```diff
@@ -0,0 +1,197 @@
...
+    });
+});
+
+
+
+
+
+
+
+
```
Comment (deepio) [inline source/js/plugins/pixel-wrapper.test.js]: Can you delete this whitespace? Careful not to commit this in the future.

File: static/js/manipulation.js [hunk 11]
```diff
@@ -478,10 +478,8 @@
 	        }
 
 	    var fkeys = Object.keys(_filterQueue);
-	    var flen = fkeys.length;
-	    var flenScale = 1 / flen;
 
-	    fkeys.map(function (filt) {});
+	    fkeys.map(function () {});
```
Comment (deepio) [inline static/js/manipulation.js]: This again, same as in here:
https://github.com/DDMAL/pixel_wrapper/blob/6b769fb917430992add15258e43445c65865e10a/source/js/plugins/_filters.js#L19-L21

File: tests/runner.html [hunk 12]
```diff
@@ -1,28 +0,0 @@
...
-    <script>document.write('<script src="http://' + (location.host || 'localhost').split(':')[0] + ':35729/livereload.js?snipver=1"></' + 'script>')</script>
-
-    <!-- Load the built library -->
-    <script src='../build/js/diva.js'></script>
-</head>
-<body>
-<!-- Required for browser reporter -->
-<div id='mocha'></div>
-</body>
-</html>
```
Comment (deepio) [inline tests/runner.html]: Was this test runner tried at all?

File: README.md (hunk 6 above)
Comment (EricHanLiu) [reply README.md]: Not anymore! I'm moving the Pixel.js job to a separate server

File: gulpfile.js (hunk 7 above)
Comment (EricHanLiu) [reply gulpfile.js]: This was (if I remember correctly) an error with strict mode and not allowing `let` to be used

File: source/js/plugins/pixel-wrapper.js (hunk 9 above)
Comment (EricHanLiu) [reply source/js/plugins/pixel-wrapper.js]: `numberInputLayers` is defined in the index.html, so the variable doesn't exist and throws the JSHint error.

File: tests/runner.html (hunk 12 above)
Comment (EricHanLiu) [reply tests/runner.html]: Nope

File: static/js/manipulation.js (hunk 11 above)
Comment (EricHanLiu) [reply static/js/manipulation.js]: This is the compiled / webpacked file so it's actually the same line as in `_filters.js`

File: .travis.yml (hunk 5 above)
Comment (EricHanLiu) [reply .travis.yml]: ok, will remove

File: source/js/plugins/pixel-wrapper.test.js (hunk 10 above)
Comment (EricHanLiu) [reply source/js/plugins/pixel-wrapper.test.js]: Done

**Comment by EricHanLiu** [comment]
Okay all the changes have been made, tests all pass now and no longer require the Rodan server to be running. removed the travis environment variables as well

**Comment by EricHanLiu** [comment]
can I merge this

**Pull request merged**